feat: expose serverStreamingCallable to save transformation

rahulKQL opened the pull request:
In Progress

rahulKQL committed:
feat: expose StreamingCallable with ReadRowsRequest in EnhancedBigtableStub

This commit would enable the user to target the table using absolute resource name on each read request. Currently we expose `ServerStreamingCallable<Query, RowT>`, which does not have an option to provide different `app-profile-id` on each request.

rahulKQL commented on google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java:
```diff
@@ -230,6 +230,66 @@ private EnhancedBigtableStub(
+   *
+   * <p>NOTE: the caller is responsible for adding tracing & metrics.
+   */
+  private <ReqT, RowT> ServerStreamingCallable<ReadRowsRequest, RowT> createReadRowsRawCallable(
+      ServerStreamingCallSettings<ReqT, Row> readRowsSettings, RowAdapter<RowT> rowAdapter) {
```
Do you think is it acceptable for a private method to have two generic parameters but return the third class instead?

Because those generics classes would be only used for fetching the settings.

dmitry-fa replied:
I don't see any problems with that

rahulKQL replied:
Thanks a lot for your opinion!!!

dmitry-fa commented on google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java:
```diff
@@ -230,6 +230,66 @@ private EnhancedBigtableStub(
+   *   <li>Upon receiving the response stream, it will merge the {@link
+   *       com.google.bigtable.v2.ReadRowsResponse.CellChunk}s in logical rows. The actual row
+   *       implementation can be configured in by the {@code rowAdapter} parameter.
+   *   <li>Retry/resume on failure.
```
be configured in by
->
be configured by

(3 times)

rahulKQL committed:
chore: reorganized createReadRows callable methods

With this commit, the public createReadRowsCallable() would refer to single createReadRowsBaseCallable.
